### DISCUSSION TITLE
[Bug]: Default value for Select/Dropdown should be first item if the intended value isn't in the list

### DISCUSSION THREAD

**Discussion by avispeng** [react-hook-form/react-hook-form#6599] [Q&A] [answered]
### Version Number

7.15.4

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-reset-form-forked-4575z?file=/src/index.js

### Steps to reproduce

1. Click 'Submit', will see {select1: "a", select2: "2"} in console. The values were set in reset() in useEffect().
2. Open select1 and select 'b', will see 'b' shown in select1 and '3' shown in select2. The available options of select2 change upon selection of select1.
3. Click 'Submit', will see {select1: "b", select2: "2"} in console.

### Expected behaviour

After the third step above, {select1: "b", select2: "3"} is expected in console.

### What browsers are you seeing the problem on?

Firefox, Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by rodrigofuentes** [comment, marked as answer]
This is not a RHF bug. You are trying to render an options field based on the value of an element that doesn't exist yet. Your conditional also doesn't take this into account. Pay close attention to what is actually happening in the component. This is why without your reset useEffect the options (3,4) is rendered.

You're doing something reactive based on select1, but not select2. You can fix your specific bug with a simple useEffect that looks like this:

https://codesandbox.io/s/react-hook-form-reset-form-forked-houmq?file=/src/index.js

Hope this helps you.

**Comment by bluebill1049** [comment]
thank you @rodrigofuentes 🙏 for the help.

**Reply by avispeng** [reply]
Thanks for the reply from @rodrigofuentes . I was reporting it as a bug because the same code worked in v6. (https://codesandbox.io/s/react-hook-form-reset-form-forked-vfciw?file=/src/index.js ). I found the behavior changed while migrating from v6 to v7. I thought it is a legitimate usecase in a dynamic form and the expected behavior is to only submit selected value if it's in the option list, otherwise submit the first item. Please let me know if the current behavior is indeed intended. Thanks.
